windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -308,13 +309,121 @@ UwbDeviceConnector::HandleNotifications(wil::shared_hfile handleDriver, std::sto
         const UWB_NOTIFICATION_DATA& notificationData = *reinterpret_cast<UWB_NOTIFICATION_DATA*>(std::data(uwbNotificationDataBuffer));
         auto uwbNotificationData = UwbCxDdi::To(notificationData);
 
-        // Invoke callback with notification data.
-        onNotification(std::move(uwbNotificationData));
+        // Invoke callbacks with notification data.
+        // Handle the notification in a fire-and-forget fashion. This may change
+        // later. Since std::async returns a future, and the future's
+        // destructor waits for it to complete, we cannot just ignore the
+        // returned future. To work around this, we move the returned future
+        // into a shared_ptr, then pass this by value to the std::async's
+        // lambda, increasing its reference count. This will ensure the future
+        // is automatically destructed once the async lambda has completed.
+        auto notificationHandlerFuture = std::make_shared<std::future<void>>();
+        *notificationHandlerFuture = std::async(std::launch::async, [this, notificationHandlerFuture, uwbNotificationData = std::move(uwbNotificationData)]() {
+            DispatchCallbacks(std::move(uwbNotificationData));
+        });
     }
 }
 
+void
+UwbDeviceConnector::OnSessionMulticastListStatus(::uwb::protocol::fira::UwbSessionUpdateMulicastListStatus statusMulticastList)
+{
+    uint32_t sessionId = statusMulticastList.SessionId;
+    auto it = m_sessionEventCallbacks.find(sessionId);
+    if (it == std::end(m_sessionEventCallbacks)) {
+        PLOG_WARNING << "Ignoring MulticastListStatus event due to missing session callback";
+        return;
+    }
+
+    auto callbacks = it->second.lock();
+    if (not(callbacks->OnSessionMembershipChanged)) {
+        PLOG_WARNING << "Ignoring MulticastListStatus event due to missing session callback";
+        return;

```

Comment:
In this case, we should remove the callback from `m_sessionEventCallbacks` since it can never be valid again. `m_sessionEventCallbacks.erase(it);` should do the trick.